targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/trng_api.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2017-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "trng_api.h"
+#include "tztrng.h"
+
+#define BITS_PER_BYTE 8
+/* Base address of the TRNG peripheral */
+#define TRNG_BASE     0x4000F000
+
+/**
+ * \defgroup hal_trng TRNG hal functions
+ * @{
+ */
+
+
+/** Initialize the TRNG peripheral
+ *
+ * @param obj The TRNG object
+ */
+void trng_init(trng_t *obj)
+{
+    /* Initialization is made in the driver */
+    (void)obj;
+}
+
+/** Deinitialize the TRNG peripheral
+ *
+ * @param obj The TRNG object
+ */
+void trng_free(trng_t *obj)
+{
+    /* The current implementation only contains a dummy version of TRNG */
+    (void)obj;
+}
+
+/** Get random data from TRNG peripheral
+ *
+ * @param obj The TRNG object
+ * @param output The pointer to an output array
+ * @param length The size of output data, to avoid buffer overwrite
+ * @param output_length The length of generated data
+ * @return 0 success, -1 fail
+ */
+int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
```

Comment:
Why are we including here doxygen (they are from header file copy paste?) ? If yes, we shall remove them. If there is another info that should be added here, could do via own implementation comments.

Reply:
They are indeed copied from the header file. I am going to remove them.